### PR TITLE
Add test summary

### PR DESCRIPTION
**Pull request by tahirmt**
Adds changes from https://github.com/fastlane-community/trainer/pull/40 and adds the total number of tests and failures to environment variables. 
Also fixes the format of the junit file to include line number

### CODE REVIEW

File: fastlane-plugin-trainer/lib/fastlane/plugin/trainer/actions/trainer_action.rb [hunk 1]
```diff
@@ -9,8 +9,20 @@ def self.run(params)
...
+        resulting_paths.values.each { |summary|
+          number_of_failures += summary.number_of_failures
+          number_of_tests += summary.number_of_tests
+        }
+
+        ENV["TRAINER_NUMBER_OF_TESTS"] = number_of_tests.to_s
+        ENV["TRAINER_NUMBER_OF_FAILURES"] = number_of_failures.to_s
+
+        resulting_paths.each do |path, summary|
+          UI.test_failure!("Unit tests failed") if fail_build && !summary.tests_successful
```
Comment (ketenshi) [inline fastlane-plugin-trainer/lib/fastlane/plugin/trainer/actions/trainer_action.rb:25]: Should this be an `||` instead?

Comment (tahirmt) [reply fastlane-plugin-trainer/lib/fastlane/plugin/trainer/actions/trainer_action.rb:25]: No the fail_build is set externally to turn off allowing the build to fail even if there are test failures

Comment (ketenshi) [reply fastlane-plugin-trainer/lib/fastlane/plugin/trainer/actions/trainer_action.rb:25]: ah ok

**Review by ketenshi — APPROVED** [review]

**Comment by kikettas** [comment]
Any plan on merging these fixes in the original repo?

**Comment by tahirmt** [comment]
> Any plan on merging these fixes in the original repo?

I didn’t open a PR there because I merged an open PR from the original repo that was open for two years and added some stuff to it here.

**Comment by kikettas** [comment]
But it fixes the missing failed tests, right? I'm wondering if the original repo can take advantage of them.

**Comment by tahirmt** [comment]
@kikettas yeah but it doesn’t look like the original repo is maintained.

**Comment by kikettas** [comment]
I see. It's a pity then, as it's the probably the best extracting tool for `xcresult` files.

**Pull request merged**